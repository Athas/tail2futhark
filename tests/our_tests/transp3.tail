let v1:[int]3 = reshape{[int],[1,3]}([2,2,2],[1,2,3,4,5,6,7,8]) in
let v2:[int]3 = transp{[int],[3]}(v1) in
let v3:[int]3 = reshape{[int],[1,3]}([2,2,2],[1,5,3,7,2,6,4,8]) in
let v6:[bool]3 = zipWith{[int,int,bool],[3]}(eqi,v2,v3) in
let v10:[int]2 = reduce{[int],[2]}(addi,0,each{[bool,int],[3]}(b2i,v6)) in
let v13:[int]1 = reduce{[int],[1]}(addi,0,v10) in
let v16:[int]0 = reduce{[int],[0]}(addi,0,v13) in
i2d(v16)
