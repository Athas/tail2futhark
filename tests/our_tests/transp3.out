fun real main() =
  toReal(reduce(+,0,[size(0,transpose(reshape((2,2,2),reshape1int(2 * 2 * 2 * 1,reshape((size(0,iota(8)) * 1),iota(8)))))),size(1,transpose(reshape((2,2,2),reshape1int(2 * 2 * 2 * 1,reshape((size(0,iota(8)) * 1),iota(8)))))),size(2,transpose(reshape((2,2,2),reshape1int(2 * 2 * 2 * 1,reshape((size(0,iota(8)) * 1),iota(8))))))]))